inspirehep/modules/disambiguation/core/ml/models.py

```diff
@@ -65,14 +82,300 @@ def save_model(self, output_filename):
+                            dtype=np.float32,
+                            decode_error='replace',
+                        )),
+                    ]),
+                    groupby=group_by_signature,
+                )),
+                ('combiner', CosineSimilarity()),
+            ])),
+            ('collaborations_similarity', Pipeline([
+                ('pairs', PairTransformer(
+                    element_transformer=Pipeline([
+                        ('collaborations', FuncTransformer(func=get_collaborations)),
+                        ('shaper', Shaper(newshape=(-1,))),
+                        ('tf-idf', TfidfVectorizer(
+                            dtype=np.float32,
+                            decode_error='replace',
+                        )),
+                    ]),
+                    groupby=group_by_signature,
+                )),
+                ('combiner', CosineSimilarity()),
+            ])),
+            ('subject_similairty', Pipeline([
+                ('pairs', PairTransformer(
+                    element_transformer=Pipeline([
+                        ('keywords', FuncTransformer(func=get_topics)),
+                        ('shaper', Shaper(newshape=(-1))),
+                        ('tf-idf', TfidfVectorizer(
+                            dtype=np.float32,
+                            decode_error='replace',
+                        )),
+                    ]),
+                    groupby=group_by_signature,
+                )),
+                ('combiner', CosineSimilarity()),
+            ])),
+            ('title_similarity', Pipeline([
+                ('pairs', PairTransformer(
+                    element_transformer=Pipeline([
+                        ('title', FuncTransformer(func=get_title)),
+                        ('shaper', Shaper(newshape=(-1,))),
+                        ('tf-idf', TfidfVectorizer(
+                            analyzer='char_wb',
+                            ngram_range=(2, 4),
+                            dtype=np.float32,
+                            decode_error='replace',
+                        )),
+                    ]),
+                    groupby=group_by_signature,
+                )),
+                ('combiner', CosineSimilarity()),
+            ])),
+            ('author_ethnicity', Pipeline([
+                ('pairs', PairTransformer(
+                    element_transformer=Pipeline([
+                        ('name', FuncTransformer(func=get_author_full_name)),
+                        ('shaper', Shaper(newshape=(-1,))),
+                        ('classifier', EstimatorTransformer(self.ethnicity_estimator.estimator)),
+                    ]),
+                    groupby=group_by_signature,
+                )),
+                ('sigmoid', FuncTransformer(func=expit)),
+                ('shaper', Shaper(newshape=(-1, 2))),
+                ('combiner', ElementMultiplication()),
+                ('reshaper', Shaper(newshape=(-1, 7))),
```

Comment:
This change did in fact become necessary because of the introduction of the previous `shaper`. I just tested locally that we can in fact get rid of both.